sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/ApproximatePercentile.scala

```diff
@@ -118,17 +118,44 @@ case class ApproximatePercentile(
     val defaultCheck = super.checkInputDataTypes()
     if (defaultCheck.isFailure) {
       defaultCheck
-    } else if (!percentageExpression.foldable || !accuracyExpression.foldable) {
-      TypeCheckFailure(s"The accuracy or percentage provided must be a constant literal")
+    } else if (!percentageExpression.foldable) {
+      DataTypeMismatch(
+        errorSubClass = "MUST_BE_CONSTANT",
```

Comment:
In fact, the input must be a foldable expr. Can't you re-use the existing error sub-class `NON_FOLDABLE_INPUT`? Maybe, make it more general to be able to use it here.

Reply:
OK, Let me see how to make `NON_FOLDABLE_INPUT ` more general